### PR TITLE
[21] [UI] As a user, I can see search bar on top

### LINKED ISSUE #21: [Integrate] As a user, I can see search bar on top
## Why

- After the user finished the search action.

## Acceptance Criteria

- In case there are more product pages => show them
- In case there are no more product pages => do not show

## Resources

Screen Shot 2021-05-04 at 11 12 13

- [Design](https://www.figma.com/proto/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=41%3A148&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by nmint8m**
https://github.com/nimblehq/git-template/issues/21

## What happened 👀

After the user finished the search action:
- In case there are more product pages => show them
- In case there are no more product pages => do not show
 
## Insight 📝

- In this PR, I only show the search bar on the top of the Search screen
- Integrate filter

### CODE REVIEW

File: ecommerce-ios/Views/SearchBar/SearchBarView.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                            label: { Image("icon-x-circle") }
+                        )
+                        .hidden(searchKeyword.isEmpty)
+                    }
+            }
+            .background(Color.galleryGray)
+            .cornerRadius(10.0)
+            Button("Cancel", action: {
+                print("did tap cancel button")
+            })
```
Comment (vnntsu) [inline ecommerce-ios/Views/SearchBar/SearchBarView.swift]:
```suggestion
            Button("Cancel")  {
                print("did tap cancel button")
            }
```

File: ecommerce-ios.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -59,6 +62,9 @@
 
 /* Begin PBXFileReference section */
 		0578CFAAA271B29A3AEF6C65 /* Pods-ecommerce-ios.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-ecommerce-ios.debug.xcconfig"; path = "Target Support Files/Pods-ecommerce-ios/Pods-ecommerce-ios.debug.xcconfig"; sourceTree = "<group>"; };
+		222EB87D26525D2B00BF46BD /* SearchBarView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = SearchBarView.swift; sourceTree = "<group>"; };
+		222EB885265268BA00BF46BD /* View+CustomOverlayView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "View+CustomOverlayView.swift"; sourceTree = "<group>"; };
+		222EB88D2652723B00BF46BD /* String+Ultility.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "String+Ultility.swift"; sourceTree = "<group>"; };
```
Comment (blyscuit) [inline ecommerce-ios.xcodeproj/project.pbxproj]: String+Utility ?

File: ecommerce-ios/Views/SearchBar/SearchBarView.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            HStack {
+                Image("icon-search")
+                    .frame(width: 14.0, height: 14.0)
+                    .padding()
+
+                TextField(
+                    "Search",
+                    text: $searchKeyword,
+                    onEditingChanged: { _ in },
+                    onCommit: { }
```
Comment (blyscuit) [inline ecommerce-ios/Views/SearchBar/SearchBarView.swift]: no need to include if not used?

File: ecommerce-ios.xcodeproj/project.pbxproj (hunk 2 above)
Comment (nmint8m) [reply ecommerce-ios.xcodeproj/project.pbxproj]: It's like a helper actually. 

image

Comment (blyscuit) [reply ecommerce-ios.xcodeproj/project.pbxproj]: I think the class name is misspelled

File: ecommerce-ios.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -273,6 +305,8 @@
 			isa = PBXGroup;
 			children = (
 				2282E9C62652320500340FEE /* View+CustomNavigationBarLargeTitle.swift */,
+				222EB885265268BA00BF46BD /* View+CustomOverlayView.swift */,
+				90248437264D16D0008ECEE5 /* View+NavigationBar.swift */,
```
Comment (vnntsu) [inline ecommerce-ios.xcodeproj/project.pbxproj]: Please remove this one, I removed it before

File: ecommerce-ios/Extensions/Color/Color+Application.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
+//
+//  Color+Application.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 13/05/2021.
+//
+
+import SwiftUI
+
+extension Color {
```
Comment (vnntsu) [inline ecommerce-ios/Extensions/Color/Color+Application.swift]: This file should be removed right?

File: ecommerce-ios/Extensions/Color/UIColor+Application.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
+//
+//  UIColor+Application.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 13/05/2021.
+//
+
+import UIKit
+
+extension UIColor {
```
Comment (vnntsu) [inline ecommerce-ios/Extensions/Color/UIColor+Application.swift]: 🤔  remove it also

File: ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift [hunk 7]
```diff
@@ -10,7 +10,7 @@ import SwiftUI
 public extension View {
 
     func navigationBarLargeTitle<Content>(@ViewBuilder customView: () -> Content) -> some View where Content: View {
-        overlay(NavigationBarLargeTiltleRepresenting(customView: customView()))
+        overlay(NavigationBarLargeTiltleRepresenting(customView: customView()).frame(width: 100, height: 100))
```
Comment (vnntsu) [inline ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift]:
```suggestion
        overlay(NavigationBarLargeTiltleRepresenting(customView: customView())
            .frame(width: 100, height: 100))
```

File: ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift [hunk 8]
```diff
@@ -52,7 +52,7 @@ extension NavigationBarLargeTiltleRepresenting {
             guard
                 let navigationBar = navigationController?.navigationBar,
                 let LargeTitleViewClass = NSClassFromString("_UINavigationBarLargeTitleView"),
-                let largeTitleView = navigationBar.subviews.first(where: { $0.isKind(of: LargeTitleViewClass.self) })
+                let largeTitleView = navigationBar.subviews.first(where: { $0.isKind(of:  LargeTitleViewClass.self) })
```
Comment (vnntsu) [inline ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift]:
```suggestion
                let largeTitleView = navigationBar.subviews.first(where: { $0.isKind(of: LargeTitleViewClass.self) })
```

File: ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift~HEAD [hunk 9]
```diff
@@ -0,0 +1,73 @@
+//
+//  View+CustomNavigationBarLargeTitle.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 17/05/2021.
+//
+
+import SwiftUI
+
+public extension View {
```
Comment (vnntsu) [inline ecommerce-ios/Extensions/View+NavigationBar/View+CustomNavigationBarLargeTitle.swift~HEAD]: ?

File: ecommerce-ios/Sources/Search/SearchScreen.swift [hunk 10]
```diff
@@ -26,6 +29,12 @@ struct SearchScreen: View {
         return vms
     }()
 
+    private var searchResultCellViewModels: [SearchItemCellViewModel] {
+        return searchKeyword.trimmed.isEmpty ? cellVỉewModels : cellVỉewModels.filter {
```
Comment (vnntsu) [inline ecommerce-ios/Sources/Search/SearchScreen.swift]:
```suggestion
        searchKeyword.trimmed.isEmpty ? cellVỉewModels : cellVỉewModels.filter {
```

File: ecommerce-ios/Views/CustomNavigationBar/ContentView/CustomNavigationBarContentView.swift [hunk 11]
```diff
@@ -32,7 +32,6 @@ struct CustomNavigationBarContentView_Previews: PreviewProvider {
             leadingView: Button("Back", action: { }).font(.footnote),
             contentView: Text("Some title").font(.title),
             trailingView: Button("Cancel", action: { }).font(.footnote)
-        )
-        .frame(width: screenWidth, height: 100.0)
+        ).frame(width: screenWidth, height: 100.0)
```
Comment (vnntsu) [inline ecommerce-ios/Views/CustomNavigationBar/ContentView/CustomNavigationBarContentView.swift]:
```suggestion
        )
        .frame(width: screenWidth, height: 100.0)
```

File: ecommerce-ios/Views/CustomNavigationBar/ContentView/CustomNavigationBarContentView.swift~HEAD [hunk 12]
```diff
@@ -0,0 +1,38 @@
+//
+//  CustomNavigationBarContentView.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 13/05/2021.
+//
+
+import SwiftUI
+
+struct CustomNavigationBarContentView<LeadingView, ContentView, TrailingView>: View
```
Comment (vnntsu) [inline ecommerce-ios/Views/CustomNavigationBar/ContentView/CustomNavigationBarContentView.swift~HEAD]: ?

**Review by blyscuit — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]
LGTM!

**Pull request merged**